roles/kubernetes-apps/ansible/tasks/coredns.yml

```diff
@@ -11,15 +11,29 @@
     - { name: coredns, file: coredns-deployment.yml, type: deployment }
     - { name: coredns, file: coredns-sa.yml, type: sa }
     - { name: coredns, file: coredns-svc.yml, type: svc }
+  register: coredns_manifests
+  vars:
+    clusterIP: "{{ skydns_server }}"
+  when:
+    - dns_mode in ['coredns', 'coredns_dual']
+    - inventory_hostname == groups['kube_control_plane'][0]
+  tags:
+    - coredns
+
+- name: Kubernetes Apps | Lay Down DNS Autoscaler templates
+  template:
+    src: "{{ item.file }}.j2"
+    dest: "{{ kube_config_dir }}/{{ item.file }}"
+    mode: 0644
+  loop:
     - { name: dns-autoscaler, file: dns-autoscaler.yml, type: deployment }
     - { name: dns-autoscaler, file: dns-autoscaler-clusterrole.yml, type: clusterrole }
     - { name: dns-autoscaler, file: dns-autoscaler-clusterrolebinding.yml, type: clusterrolebinding }
     - { name: dns-autoscaler, file: dns-autoscaler-sa.yml, type: sa }
-  register: coredns_manifests
-  vars:
-    clusterIP: "{{ skydns_server }}"
+  register: dns_autoscaler_manifests
   when:
     - dns_mode in ['coredns', 'coredns_dual']
+    - enable_dns_autoscaler
```

Comment:
Instead of adding a tasks, how about using something like `enable_dns_autoscaler or name != dns-autoscaler` ?

Reply:
> Instead of adding a tasks, how about using something like `enable_dns_autoscaler or name != dns-autoscaler` ?

@VannTen  done, PTAL